src/android/Capture.java

```diff
@@ -250,13 +304,17 @@ private static void createWritableFile(File file) throws IOException {
      * Sets up an intent to capture video.  Result handled by onActivityResult()
      */
     private void captureVideo(int duration, int quality) {
-        Intent intent = new Intent(android.provider.MediaStore.ACTION_VIDEO_CAPTURE);
+        if(cameraPermissionInManifest && !PermissionHelper.hasPermission(this, Manifest.permission.CAMERA)) {
+            PermissionHelper.requestPermission(this, CAPTURE_VIDEO_SEC, Manifest.permission.CAMERA);
+        } else {
+            Intent intent = new Intent(android.provider.MediaStore.ACTION_VIDEO_CAPTURE);
 
-        if(Build.VERSION.SDK_INT > 7){
-            intent.putExtra("android.intent.extra.durationLimit", duration);
-            intent.putExtra("android.intent.extra.videoQuality", quality);
+            if(Build.VERSION.SDK_INT > 7){
```

Comment:
Oh, wow, I think I glanced over that! I agree that we should remove it. I'll update the PR